apps/pre-award/lambdas/application-deadline-reminder/incomplete_application.py

```diff
@@ -0,0 +1,216 @@
+from datetime import datetime
+from dateutil import tz
+from test_config import Config
+from data import get_data_safe
+
+import requests
+
+import logging
+
+logging.getLogger("lambda_runtime").setLevel(logging.INFO)
+logging.getLogger().setLevel(logging.DEBUG)
+
+
+def process_events():
+    """
+    Pulls events from the fund store and checks if they need processing. If so, the relevant processor
+    will be called (determined by event type). If the processing was successful, the event is updated
+    and marked as processed.
+
+    Return:
+        True if the function ran without issue.
+    """
+    logging.info("Running event check")
+    uk_timezone = tz.gettz("Europe/London")
+    current_datetime = datetime.now(uk_timezone).replace(tzinfo=None)
+    funds_response = requests.get(Config.FUND_STORE_API_HOST + Config.FUNDS_ENDPOINT)
+    # If we cannot retrieve the funds then fail loudly
+    funds_response.raise_for_status()
+    funds = funds_response.json()
+
+    # Iterate over rounds and events. Note that failure to retrieve rounds / events should be non blocking so that
+    # the rest of the rounds / events can still be processed
+    for fund in funds:
+        fund_id = fund["id"]
+        fund_info = get_data_safe(
+            Config.FUND_STORE_API_HOST + Config.FUND_ENDPOINT.format(fund_id=fund_id)
+        )
+        rounds = get_data_safe(
+            Config.FUND_STORE_API_HOST
+            + Config.FUND_ROUNDS_ENDPOINT.format(fund_id=fund_id)
+        )
+        if fund_info and rounds:
+            fund_name = fund_info["name"]
+
+            for round in rounds:
+                round_id = round["id"]
+                round_name = round["title"]
+                round_contact_email = round.get("contact_email")
+                events = get_data_safe(
+                    Config.FUND_STORE_API_HOST
+                    + Config.FUND_EVENTS_ENDPOINT.format(
+                        fund_id=fund_id, round_id=round_id
+                    )
+                )
+
+                if events:
```

Comment:
can we use statement like this: 
if not events:
   continue
for event in events:
      event_type = event["type"]
      event_activation_date = datetime.strptime(
          event.get("activation_date"), "%Y-%m-%dT%H:%M:%S"
      )
      event_id = event["id"]
                        event_processed = event["processed"]

Reply:
which will help us in getting of nested If's

Reply:
Good idea! Updated.